planning/behavior_path_planner/autoware_behavior_path_dynamic_obstacle_avoidance_module/include/autoware/behavior_path_dynamic_obstacle_avoidance_module/scene.hpp

```diff
@@ -350,6 +359,7 @@ class DynamicObstacleAvoidanceModule : public SceneModuleInterface
     parameters_ = std::any_cast<std::shared_ptr<DynamicAvoidanceParameters>>(parameters);
   }
 
+  bool arrived_path_end_{false};
```

Comment:
If we can separate this feature from the drivable area expansion, I suggest to make separate PR.

Reply:
I had to change the contents of canTransitSuccessState so that the expansion would stop at the right time. But I can create a PR for that part if you want.

```
const auto & data = getPreviousModuleOutput();
  const auto idx = planner_data_->findEgoIndex(data.reference_path.points);
  if (idx == data.reference_path.points.size() - 1) {
    arrived_path_end_ = true;
  }
  constexpr double THRESHOLD = 2.0;
  const auto is_further_than_threshold =
    autoware::universe_utils::calcDistance2d(
      getEgoPose(), autoware::universe_utils::getPose(data.reference_path.points.back())) >
    THRESHOLD;
  if (is_further_than_threshold && arrived_path_end_) {
    RCLCPP_WARN(getLogger(), "Reach path end point. Exit.");
    return true;
  }
```